### PR TITLE
Guptavertika/release network microsoft.network 2024 03 01

### PR DESCRIPTION
**Pull request by GuptaVertika**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    - _edit this with your clarification_ This PR is combination of Multiple Networking Partner PR, Which is already approved by ARM reviewer before hand.
    - https://github.com/Azure/azure-rest-api-specs/pull/30469
    - https://github.com/Azure/azure-rest-api-specs/pull/30386
    - https://github.com/Azure/azure-rest-api-specs/pull/30364
    - https://github.com/Azure/azure-rest-api-specs/pull/30342
    - https://github.com/Azure/azure-rest-api-specs/pull/30332
    - https://github.com/Azure/azure-rest-api-specs/pull/30196
    - https://github.com/Azure/azure-rest-api-specs/pull/30176
    - https://github.com/Azure/azure-rest-api-specs/pull/30102
    - https://github.com/Azure/azure-rest-api-specs/pull/30096
    - https://github.com/Azure/azure-rest-api-specs/pull/30083
    - https://github.com/Azure/azure-rest-api-specs/pull/30074
    - https://github.com/Azure/azure-rest-api-specs/pull/30071
    - https://github.com/Azure/azure-rest-api-specs/pull/30044
    - https://github.com/Azure/azure-rest-api-specs/pull/30028
    - https://github.com/Azure/azure-rest-api-specs/pull/29988
    - https://github.com/Azure/azure-rest-api-specs/pull/29986
    - https://github.com/Azure/azure-rest-api-specs/pull/29969
    - https://github.com/Azure/azure-rest-api-specs/pull/29933
    - https://github.com/Azure/azure-rest-api-specs/pull/29868
    - https://github.com/Azure/azure-rest-api-specs/pull/29855
    - https://github.com/Azure/azure-rest-api-specs/pull/29854
    - https://github.com/Azure/azure-rest-api-specs/pull/29833
    - https://github.com/Azure/azure-rest-api-specs/pull/29830
    - https://github.com/Azure/azure-rest-api-specs/pull/29826
    - https://github.com/Azure/azure-rest-api-specs/pull/29759
    - https://github.com/Azure/azure-rest-api-specs/pull/29754
    - https://github.com/Azure/azure-rest-api-specs/pull/29727
    - https://github.com/Azure/azure-rest-api-specs/pull/29715

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by razvanbadea-msft** [comment]
Fix the required Swagger PrettierCheck , Swagger ModelValidation,  if there are Swagger LintDiff errors to suppress add some comments why you need them

**Comment by ramoka178** [comment]
There are Swagger LintDiff and Swagger ModelValidation failed checks. 
Please fix them. 

If they are previously approved on 2024-03-01 on your earlier PRs to the release branch, please provide those PR links to review.

**Comment by kamboj-prjwl** [comment]
> There are Swagger LintDiff and Swagger ModelValidation failed checks. Please fix them.
> 
> If they are previously approved on 2024-03-01 on your earlier PRs to the release branch, please provide those PR links to review.

For LintDiff violation, I have raised another PR which is approved & merged in this branch itself. 
PR: https://github.com/Azure/azure-rest-api-specs/pull/30386
Can you please provide the Suppress-Approved .

**Comment by kamboj-prjwl** [comment]
> There are Swagger LintDiff and Swagger ModelValidation failed checks. Please fix them.
> 
> If they are previously approved on 2024-03-01 on your earlier PRs to the release branch, please provide those PR links to review.

Can you please provide model validation approval on this PR.  While we are aware that there are still model validation failures, we have significantly reduced the count from over 450+ to below 300.
Moreover, our partner teams are actively working to resolve the remaining issues.

File: specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/virtualNetworkGateway.json [hunk 1]
```diff
@@ -4284,8 +4284,7 @@
         "ECP256",
         "ECP384",
         "PFS24",
-        "PFS14",
-        "PFSMM"
+        "PFS14"
```
Comment (TimLovellSmith) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/virtualNetworkGateway.json:4287]: >PFS14 [](http://example.com/codeflow?start=9&length=5)

is removing PSMM intended? non-breaking?

File: specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/azureFirewall.json [hunk 2]
```diff
@@ -0,0 +1,1350 @@
...
+          "minimum": 2,
+          "description": "The minimum number of capacity units for this azure firewall. Use null to reset the value to the service default.",
+          "x-nullable": true
+        },
+        "maxCapacity": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 2,
+          "description": "The maximum number of capacity units for this azure firewall. Use null to reset the value to the service default.",
+          "x-nullable": true
```
Comment (TimLovellSmith) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/azureFirewall.json:1344]: >nullable [](http://example.com/codeflow?start=13&length=8)

x-nullable, or representation with explicit nulls, is not generally recommended in Azure APIs.
Is there a reason you need it??

File: specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/azureFirewall.json [hunk 3]
```diff
@@ -0,0 +1,1350 @@
...
+    },
+    "AzureFirewallAutoscaleConfiguration": {
+      "type": "object",
+      "properties": {
+        "minCapacity": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 2,
+          "description": "The minimum number of capacity units for this azure firewall. Use null to reset the value to the service default.",
+          "x-nullable": true
```
Comment (TimLovellSmith) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/azureFirewall.json:1337]: >   "x-nullable": true [](http://example.com/codeflow?start=7&length=21)

x-nullable, or representation with explicit nulls, is not generally recommended in Azure APIs.
Is there a reason you need it??

**Comment by TimLovellSmith** [comment]
You'll still need to add suppressions for suppression review, or fix the causes of errors that LintDiff and ModelValidation are raising

"[must fix]The following errors/warnings are introduced by current PR:"

**Comment by kamboj-prjwl** [comment]
> You'll still need to add suppressions for suppression review, or fix the causes of errors that LintDiff and ModelValidation are raising
> 
> "[must fix]The following errors/warnings are introduced by current PR:"

We are looking for model validation approval tag for Swagger Release 2024-03-01. Though we are aware that there are still model validation failures, we have significantly reduced the count from over 400 to 280.
We are pushing the partner team to fix those validations also by next release.

File: specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/azureFirewall.json (hunk 3 above)
Comment (kamboj-prjwl) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/azureFirewall.json:1337]: Following changes were introduced by PR : https://github.com/Azure/azure-rest-api-specs/pull/29933/files#diff-451f3487bb10debee2bd5e66d2483e6c86e72c7d134b4431bb027cf0b9ceec38 which is already approved

File: specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/virtualNetworkGateway.json (hunk 1 above)
Comment (kamboj-prjwl) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/virtualNetworkGateway.json:4287]: Yes, it is intended. This merged PR : [virtualNetworkGateway.json](https://github.com/Azure/azure-rest-api-specs/pull/29993/files/7432b5df05cb85550da88da3d8932ae539d5c816#diff-1735ce4b05b4eb68f09c6d8e95a0a17a8605b2b3fd65114df74bba614f39ec05) introduced the changes

**Comment by TimLovellSmith** [comment]
Many of your model validation errors look simple to fix, so my take is why not just fix them now.

**Comment by kamboj-prjwl** [comment]
> Many of your model validation errors look simple to fix, so my take is why not just fix them now. /cc @rkmanda

Can you please approve the model validation & LintDiff checks. We have reduced the count for model violation from 400+ to below 200 & partner teams are actively working to resolve the remaining by next release. However, we are approaching the release timeline for Swagger & PowerShell. Can you please provide Model violation & LintDiff approval tags.

**Review by kamboj-prjwl — APPROVED** [review]

**Pull request merged**